### PR TITLE
feat: intermediate codelab

### PR DESCRIPTION
**Pull request by engelke**
## Description
Fixes b/310286391

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: practice-folder/intermediate-sample/main_test.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    yield name
+
+    bucket.delete(force=True)
+
+
+def test_list_blobs(bucket_name, capsys):
+    blobs = main.list_blobs(bucket_name)
+
+    assert len(blobs) == 2
+    assert blobs[0] == "blob-one"
```
Comment (msampathkumar) [inline practice-folder/intermediate-sample/main_test.py:46]:
```
    assert blobs[0] == "blob-one"
```

Q: How can I be sure that order of object is ["blob-one", "blob-two", ""]

Comment (engelke) [reply practice-folder/intermediate-sample/main_test.py:46]: Good point. I'll change this to "blob-one" in blobs.

File: practice-folder/intermediate-sample/main_test.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    assert blobs[1] == "blob-two"
+
+    captured = capsys.readouterr()
```
[the remark this answers is not in the document]
Comment (engelke) [reply practice-folder/intermediate-sample/main_test.py:49]: Right now, our standards say samples should print the results. I'm not a fan, but this is very common in samples and I wanted to show an example of it here.

**Review by msampathkumar — APPROVED** [review]
LGTM

**Pull request merged**